### ISSUE TITLE
Is `export default ` HOC pattern should be flagged?

### ISSUE THREAD

**Issue by DianaSuvorova** [ArnaudBarre/eslint-plugin-react-refresh#6]
We get this type of pattern flagged by the rule:

```js
  const MainPageHeader = () => <HeaderWrapper/>
  export default withRouter(MainPageHeader);
```

error:
```
AssertionError [ERR_ASSERTION]: Should have no errors but had 1: [
  {
    ruleId: 'only-export-components',
    severity: 1,
    message: 'Fast refresh only works when a file only export components. Move your component(s) to a separate file.',
    line: 2,
    column: 15,
    nodeType: 'Identifier',
    messageId: 'localComponents',
    endLine: 2,
    endColumn: 29
  }
]
```

while this one is valid from a react-refresh perspective:

```js
      const MainPageHeader = () => <HeaderWrapper/>
      export const SubScreen =  withRouter(MainPageHeader);
 ```

**Comment by DianaSuvorova** [comment]
yes, I agree with all of the above. 

I am still trying to figure out whether by react-refresh spec it triggers a whole tree reload:

https://github.com/pmmmwh/react-refresh-webpack-plugin/issues/99#issuecomment-634143932

https://github.com/facebook/react/issues/17142#issuecomment-543996489

**Comment by ArnaudBarre** [comment]
Yes absolutely you can send a PR!

**Comment by DianaSuvorova** [comment]
Default HOC exports are actually flagged correctly with the current version of the plugin and the error message is descriptive enough. Apologies for the confusion. 

One pattern that we still use in our codebase and that wasn't flagged is a default CallExpression export: 
`compose()(MainComponent)`

Here is a PR to address that: https://github.com/ArnaudBarre/eslint-plugin-react-refresh/pull/7
